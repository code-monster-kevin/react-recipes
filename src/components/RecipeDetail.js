import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const RecipeDetail = props => {
  if (!props.recipe) {
    return (
      <p
        style={props.style}
        className={classNames('h3 p2 bg-white italic center', props.className)}
      >
        Select the recipe to see details
      </p>
    );
  }

  return (
    <div
      style={props.style}
      className={classNames('p2 bg-white', props.className)}
    >
      <h2 className="h2">{props.recipe.name}</h2>
      <img className="fit" alt={props.recipe.name} src={props.recipe.image} />
      <div>
        <span>{props.recipe.category}</span>
        <span>{props.recipe.calories}</span>
      </div>
      <h3>Ingredients</h3>
      {props.recipe.ingredients && (
        <ul>
          {props.recipe.ingredients.map(ingredient => (
            <li key={ingredient}>{ingredient}</li>
          ))}
        </ul>
      )}
      <h3>Steps</h3>
      {props.recipe.steps && (
        <ol>{props.recipe.steps.map(step => <li key={step}>{step}</li>)}</ol>
      )}
      <Link to={`/recipe/${props.recipe.id}`}>Open</Link>
    </div>
  );
};

RecipeDetail.propTypes = {
  recipe: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
};

export default RecipeDetail;
