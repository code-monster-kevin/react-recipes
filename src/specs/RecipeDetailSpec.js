import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';

describe('<RecipeDetail />', () => {
  let mockRecipe;

  beforeEach(() => {
    mockRecipe = {
      id: 1,
      name: 'Mock Recipe',
      category: 'Mock Category',
      ingredients: ['Ingredient 1', 'Ingredient 2', 'Ingredient 3'],
      steps: ['Step 1', 'Step 2', 'Step 3', 'Step 4', 'Step 5'],
    };
  });

  test('No props or state input', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Valid recipes props input', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={mockRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Valid classnames props input', () => {
    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={mockRecipe} className="classname-test" />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('No ingredients data in recipe', () => {
    delete mockRecipe.ingredients;

    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={mockRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('No steps data in recipe', () => {
    delete mockRecipe.steps;

    const component = renderer.create(
      <BrowserRouter>
        <RecipeDetail recipe={mockRecipe} />
      </BrowserRouter>,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
