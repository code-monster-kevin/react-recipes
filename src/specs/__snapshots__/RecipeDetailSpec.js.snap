// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> No ingredients data in recipe 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Mock Recipe
  </h2>
  <img
    alt="Mock Recipe"
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Mock Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
    <li>
      Step 3
    </li>
    <li>
      Step 4
    </li>
    <li>
      Step 5
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    Open
  </a>
</div>
`;

exports[`<RecipeDetail /> No props or state input 1`] = `
<p
  className="h3 p2 bg-white italic center"
  style={undefined}
>
  Select the recipe to see details
</p>
`;

exports[`<RecipeDetail /> No steps data in recipe 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Mock Recipe
  </h2>
  <img
    alt="Mock Recipe"
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Mock Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      Ingredient 1
    </li>
    <li>
      Ingredient 2
    </li>
    <li>
      Ingredient 3
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    Open
  </a>
</div>
`;

exports[`<RecipeDetail /> Valid classnames props input 1`] = `
<div
  className="p2 bg-white classname-test"
  style={undefined}
>
  <h2
    className="h2"
  >
    Mock Recipe
  </h2>
  <img
    alt="Mock Recipe"
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Mock Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      Ingredient 1
    </li>
    <li>
      Ingredient 2
    </li>
    <li>
      Ingredient 3
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
    <li>
      Step 3
    </li>
    <li>
      Step 4
    </li>
    <li>
      Step 5
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    Open
  </a>
</div>
`;

exports[`<RecipeDetail /> Valid recipes props input 1`] = `
<div
  className="p2 bg-white"
  style={undefined}
>
  <h2
    className="h2"
  >
    Mock Recipe
  </h2>
  <img
    alt="Mock Recipe"
    className="fit"
    src={undefined}
  />
  <div>
    <span>
      Mock Category
    </span>
    <span />
  </div>
  <h3>
    Ingredients
  </h3>
  <ul>
    <li>
      Ingredient 1
    </li>
    <li>
      Ingredient 2
    </li>
    <li>
      Ingredient 3
    </li>
  </ul>
  <h3>
    Steps
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
    <li>
      Step 3
    </li>
    <li>
      Step 4
    </li>
    <li>
      Step 5
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    Open
  </a>
</div>
`;
